CONTRIBUTING.md

```diff
@@ -0,0 +1,24 @@
+# Contributing
+
+Contributions to slide are warmly welcomed. Here's an overview of the code base of a slide and how to contribute!
+
+## Building
+Slide is super easy to build. Clone the repo and run cargo test to make sure your framework is setup. 
+Slide currently runs on the latest stable version of rust. As slide is in early development versions, it is currently
+not setup with a ui for applying rules without touching the codebase. 
+
+## Testing 
+Every contribution to slides must have tests. If your PR does not, tests will most likely be requested. Slide has its own 
+development tool called `ladder` that makes it easy to test for both errors and proper simplifications. It automatically
```

Comment:
```suggestion
development tool called [`ladder`](ladder.py) that makes it easy to test for both errors and proper simplifications. It automatically
```

Ladder should probably be its own section to go over the commands it has

Reply:
Here it's probably enough to say "make sure you test unit tests, and if you're writing system tests, check out the [system test documentation](slide/src/test/README.md)"